### PR TITLE
#142 Syntax check function

### PR DESCRIPTION
**Pull request by nasirhemed**
Change the warning message display for syntax check to give a clearer message 

current

### CODE REVIEW

**Comment by amgrubb** [comment]
@nasirhemed Please add the issue number to the title and description of the pull request. Thanks!

File: leaf-ui/scripts/main.js [hunk 1]
```diff
@@ -457,53 +457,64 @@ function syntaxCheck(){
...
+                subErrorText += this.getNodeName(element) + ", ";
+            });
+            errorText = errorText.substring(0, errorText.lastIndexOf(",")) + "</b>";
+            errorText += " and " + "<b style='color:blue'>" + this.getNodeName(sourceList[sourceList.length - 1]) + "</b>";
+
+
+            subErrorText = subErrorText.substring(0, subErrorText.lastIndexOf(",")) + " ";
+            subErrorText += " and " + this.getNodeName(sourceList[sourceList.length - 1]);
+            subErrorText += " to " + this.getNodeName(key);
+            // TODO: Code could be cleaned by copying substring of errorText to subErrorText
```
Comment (xorye) [inline leaf-ui/scripts/main.js]: Was there a particuliar reason why this was not done instead?

File: leaf-ui/scripts/main.js [hunk 2]
```diff
@@ -457,53 +457,64 @@ function syntaxCheck(){
...
+        }
+        else{
+            destSourceMapper[key]["findview"].forEach(function(element){
+                element.model.attr({'.connection': {'stroke': '#000000'}});
+                element.model.attr({'.marker-target': {'stroke': '#000000'}});
+                element.model.attr({'.connection': {'stroke-width': '1'}});
+                element.model.attr({'.marker-target': {'stroke-width': '1'}});
+            })
+        }
+        j+=1;
```
Comment (xorye) [inline leaf-ui/scripts/main.js]: This j variable is not necessary since the new error messages do not count the number of errors anymore.

**Comment by nasirhemed** [comment]
image

The way it currently looks like

**Comment by xorye** [comment]
Yes, it is ready to be merged.

**Comment by amgrubb** [comment]
Sweet, can you complete the review?

File: leaf-ui/scripts/main.js [hunk 3]
```diff
@@ -235,7 +235,10 @@ function saveLinks(mode){
 //Switch to analysis mode
 $('#analysis-btn').on('click', function(){
 	syntaxCheck();
-	console.log(returned_syntaxCheck());
+<<<<<<< HEAD
```
Comment (xorye) [inline leaf-ui/scripts/main.js]: Why is there a merge conflict in the checkin?
@nasirhemed

**Comment by xorye** [comment]
We have asked Alicia, and those are indeed invalid combinations. But, I just recently discovered some inconsistencies/inaccuracies with the error messages, so I am dealing with that now. I will push some new code in a few hours.

**Comment by xorye** [comment]
Ok, it is now ready to be reviewed.

**Comment by amgrubb** [comment]
This is for issue #142

**Pull request merged**